opencga-storage/opencga-storage-core/src/main/java/org/opencb/opencga/storage/core/variant/annotation/annotators/extensions/cosmic/CosmicVariantAnnotatorExtensionTask.java

```diff
@@ -0,0 +1,173 @@
+            }
+        }
+        return Collections.singletonList(dbLocation.toUri());
+    }
+
+    @Override
+    public void checkAvailable() throws IllegalArgumentException {
+        if (dbLocation == null || !Files.exists(dbLocation)) {
+            throw new IllegalArgumentException("COSMIC annotator extension is not available");
+        }
+    }
+
+    @Override
+    public ObjectMap getOptions() {
+        return options;
+    }
+
+    @Override
+    public ObjectMap getMetadata() {
+        return new ObjectMap("data", ID)
+                .append("version", cosmicVersion)
+                .append("assembly", assembly);
+    }
+
+    @Override
+    public List<VariantAnnotation> apply(List<VariantAnnotation> list) throws Exception {
+        for (VariantAnnotation variantAnnotation : list) {
+            Variant variant;
+            try {
+                variant = new Variant(variantAnnotation.getChromosome(), variantAnnotation.getStart(), variantAnnotation.getReference(),
+                        variantAnnotation.getAlternate());
+            } catch (Exception e) {
+                logger.warn("Skipping variant: " + e.getMessage());
+                continue;
+            }
+            byte[] key = variant.toString().getBytes();
+            byte[] dbContent = rdb.get(key);
+            if (dbContent != null) {
+                List<EvidenceEntry> evidenceEntryList = objectReader.readValue(dbContent);
+                if (variantAnnotation.getTraitAssociation() == null) {
+                    variantAnnotation.setTraitAssociation(evidenceEntryList);
+                } else {
+                    variantAnnotation.getTraitAssociation().addAll(evidenceEntryList);
+                }
+            }
+        }
+        return list;
+    }
+
+    @Override
+    public void post() throws Exception {
+        closeRocksDB();
+    }
+
+
+    private void closeRocksDB() {
+        if (rdb != null) {
+            rdb.close();
+        }
+        if (dbOption != null) {
+            dbOption.dispose();
+        }
+    }
+
+    private void initRockDB(boolean forceCreate) throws ToolException {
+        boolean indexingNeeded = forceCreate || !Files.exists(dbLocation);
+        // a static method that loads the RocksDB C++ library.
+        RocksDB.loadLibrary();
+        // the Options class contains a set of configurable DB options
+        // that determines the behavior of a database.
+        dbOption = new Options().setCreateIfMissing(true);
+
+        rdb = null;
+        try {
+            // a factory method that returns a RocksDB instance
+            if (indexingNeeded) {
+                rdb = RocksDB.open(dbOption, dbLocation.toAbsolutePath().toString());
+            } else {
+                rdb = RocksDB.openReadOnly(dbOption, dbLocation.toAbsolutePath().toString());
+            }
+        } catch (RocksDBException e) {
+            // Do some error handling
+            throw new ToolException("", e);
```

Comment:
No need to wrap the exception. The calling method can throw "Exception", and you're not improving the error message